MapboxNavigation/InstructionsCardCollection.swift

```diff
@@ -0,0 +1,370 @@
+import MapboxDirections
+import MapboxCoreNavigation
+
+@objc public protocol InstructionsCardCollectionDelegate: InstructionsCardContainerViewDelegate {
+    /**
+     Called when previewing the steps on the current route.
+     
+     Implementing this method will allow developers to move focus to the maneuver that corresponds to the step currently previewed.
+     - parameter instructionsCardCollection: The instructions card collection instance.
+     - parameter step: The step for the maneuver instruction in preview.
+     */
+    @objc(instructionsCardCollection:previewFor:)
+    func instructionsCardCollection(_ instructionsCardCollection: InstructionsCardCollection, previewFor step: RouteStep)
+    
+    /**
+     Offers the delegate the opportunity to customize the size of a prototype collection view cell per the associated trait collection.
+     
+     - parameter instructionsCardCollection: The instructions card collection instance.
+     - parameter cardSizeForTraitcollection: The trait collection associated to the current container view controller.
+     - returns: The preferred size of the cards for each cell in the instructions card collection.
+     */
+    @objc(instructionsCardCollection:cardSizeForTraitcollection:)
+    optional func instructionsCardCollection(_ instructionsCardCollection: InstructionsCardCollection, cardSizeForTraitcollection: UITraitCollection) -> CGSize
+}
+
+open class InstructionsCardCollection: UIViewController {
+    typealias InstructionsCardCollectionLayout = UICollectionViewFlowLayout
+    
+    var routeProgress: RouteProgress?
+    var cardSize: CGSize = .zero
+    var cardStyle: DayInstructionsCardStyle = DayInstructionsCardStyle()
+    
+    var instructionCollectionView: UICollectionView!
+    var instructionsCardLayout: InstructionsCardCollectionLayout!
+    
+    public private(set) var isInPreview = false
+    private var currentStepIndex: Int?
+    
+    var steps: [RouteStep]? {
+        guard let stepIndex = routeProgress?.currentLegProgress.stepIndex, let steps = routeProgress?.currentLeg.steps else { return nil }
```

Comment:
what about multileg routes?

Reply:
coming soon.

Reply:
@frederoni captured in this issue ticket #2176